wkcuber/wkcuber/tile_cubing.py

```diff
@@ -292,39 +292,81 @@ def tile_cubing(
         return
 
     # Determine tile size from first matching file
-    tile_size = image_reader.read_dimensions(arbitrary_file)
+    num_x, num_y = image_reader.read_dimensions(arbitrary_file)
+    num_z = max_dimensions["z"] - min_dimensions["z"] + 1
     num_channels = image_reader.read_channel_count(arbitrary_file)
     logging.info(
         "Found source files: count={} with tile_size={}x{}".format(
-            file_count, tile_size[0], tile_size[1]
+            file_count, num_x, num_y
         )
     )
     if args is None or not hasattr(args, "dtype") or args.dtype is None:
         dtype = image_reader.read_dtype(arbitrary_file)
+    else:
+        dtype = args.dtype
+
+    target_ds = Dataset.get_or_create(target_path, scale=(1, 1, 1))
+    is_segmentation_layer = layer_name == "segmentation"
+    if is_segmentation_layer:
+        target_layer = target_ds.get_or_add_layer(
+            layer_name,
+            LayerCategories.SEGMENTATION_TYPE,
+            dtype_per_channel=dtype,
+            num_channels=num_channels,
+            largest_segment_id=0,
+        )
+    else:
+        target_layer = target_ds.get_or_add_layer(
+            layer_name,
+            LayerCategories.COLOR_TYPE,
+            dtype_per_channel=dtype,
+            num_channels=num_channels,
+        )
+    if target_layer.bounding_box == BoundingBox((0, 0, 0), (0, 0, 0)):
+        # If the layer is empty, we want to set the bbox directly because extending it
+        # would mean that the bbox would always start at (0, 0, 0)
+        target_layer.bounding_box = BoundingBox(
+                Vec3Int(0, 0, min_dimensions["z"]),
+                Vec3Int(num_x, num_y, num_z),
+            )
+    else:
+        target_layer.bounding_box = target_layer.bounding_box.extended_by(
+            BoundingBox(
+                Vec3Int(0, 0, min_dimensions["z"]),
+                Vec3Int(num_x, num_y, num_z),
+            )
+        )
```

Comment:
I have set the z-offset to `min_dimensions["z"]`, but I am not sure if this is correct. The test dataset `/testdata/temca2` starts with `z=1`. Currently this would lead to a dataset with a bounding box that starts at 1. Is this intended or does the index for tiles always start at "1"?

Reply:
I think, the default `cubing` strategy writes moves the data so that it starts at `z=0`. However, I think, we should try to keep the original indexing by default (unless specified differently). This is a broader issue, though.
For this PR, reusing `z` seems okay to me. Would be a bummer, though, if this causes tests to fail :smile:

Reply:
The test dataset `temca2` starts at `z=1` and ends at `z=40`.
The new code generates a datasource-config with bounding box `topLeft=(143360, 61440, 1), size=(6144, 2024, 40)`. I have run the test with the code from the master and this PR. They both have the same data in this boundingbox (which contains indeed some data). Therefore, I conclude that the code works.
As far as I can see it, the previous code also wrote the data to the "original" index.

Reply:
Great, sounds good then :+1: